main.js

```diff
@@ -0,0 +1,173 @@
+
+
+//object to store state of comments
+const commentsObject = {
+  numberOfComments: 0,
+  comments: {}
+};
+
+//overall container to hold all comments
+const $commentContainer = $('#comment-container');
+
+
+$(document).ready(function() {
+
+  //ANCHOR creates new comment when form is submitted
+  $('#new-comment-form').submit(function(e) {
+    e.preventDefault() //prevents submission from reloading page
+
+    //each comment created adds to the numberOfComments in the commentsObject
+    commentsObject.numberOfComments++
+    const $commentNumber = commentsObject.numberOfComments
+
+    //retrieving values of Name and Message 
+    const $commentName = $(this).find('#comment-name').val()
+    const $commentMessage = $(this).find('#comment-message').val()
+
+    //populates commentObject with data on each comment
+    commentsObject.comments[`comment${$commentNumber}`] = {
+      name: $commentName,
+      message: $commentMessage,
+      likes: 0, 
+      numberOfReplies: 0,
+      replies: {}
+    }
+
+    //variable to be used to assign IDs to each post
+    const $commentId = commentsObject.comments[`comment${$commentNumber}`];
+
+    //HTML template for each new comment
+    const newComment = $('#new-comment-template').html();
+    const commentTemplate = Handlebars.compile(newComment);
+    const commentHTML = commentTemplate(
+      {
+        id: `comment${commentsObject.numberOfComments}`,
+        faker: `${faker.image.people('100', '100', true)}`,
+        name: `${$commentId.name}`,
+        message: `${$commentId.message}`,
+        date: `${getCurrentTime(new Date)}`
+      }
+    )
+
+    //comment will fade in on submission
+    $(commentHTML).hide().appendTo($commentContainer).fadeIn(200)
+
+    $(this)[0].reset() //resets form after submission
+  })
+
+
+  //ANCHOR hover over effect on like and reply
+  $($commentContainer).on('mouseenter', '.hover', function() {
+    $(this).addClass('bg-primary text-light')
+    $(this).css('cursor', 'pointer')
+  }).on("mouseleave", '.hover', function(){
+    $(this).removeClass('bg-primary text-light')
+  })
+
+
+  //ANCHOR like functionality for comments
+  $($commentContainer).on('click', '.like-button', function(e) {
+
+    const $currentCommentId = $(this).closest('.user-comment').attr('id');
+    const $currentReplyId = $(this).closest('.user-reply').attr('id');
+
+    console.log($currentCommentId, $currentReplyId)
+
+    //logic to determine whether like being clicked is comment or reply
+    if($currentReplyId) {
+      //apply likes only to select reply
+      commentsObject.comments[$currentCommentId]['replies'][$currentReplyId]['likes'] = commentsObject.comments[$currentCommentId]['replies'][$currentReplyId]['likes'] + 1 || 1
+
+      const $currentLikes = commentsObject.comments[$currentCommentId]['replies'][$currentReplyId]['likes']
+      
+      if($(this).parent().children().length === 3 ) {
```

Comment:
I wonder why we're checking for this length? Nothing wrong with it, just can't figure out exactly what's going on.